ml-agents/mlagents/model_serialization.py

```diff
@@ -67,18 +75,28 @@ def export_policy_model(
         logger.info(f"Exported {settings.model_path}.nn file")
 
     # Save to onnx too (if we were able to import it)
-    if ONNX_EXPORT_ENABLED and settings.convert_to_onnx:
-        try:
-            onnx_graph = convert_frozen_to_onnx(settings, frozen_graph_def)
-            onnx_output_path = settings.model_path + ".onnx"
-            with open(onnx_output_path, "wb") as f:
-                f.write(onnx_graph.SerializeToString())
-            logger.info(f"Converting to {onnx_output_path}")
-        except Exception:
-            logger.exception(
-                "Exception trying to save ONNX graph. Please report this error on "
-                "https://github.com/Unity-Technologies/ml-agents/issues and "
-                "attach a copy of frozen_graph_def.pb"
+    if ONNX_EXPORT_ENABLED:
+        if settings.convert_to_onnx:
+            try:
+                onnx_graph = convert_frozen_to_onnx(settings, frozen_graph_def)
+                onnx_output_path = settings.model_path + ".onnx"
+                with open(onnx_output_path, "wb") as f:
+                    f.write(onnx_graph.SerializeToString())
+                logger.info(f"Converting to {onnx_output_path}")
+            except Exception:
+                # Make conversion errors fatal depending on environment variables (only done during CI)
+                if _enforce_onnx_conversion():
+                    raise
+                logger.exception(
+                    "Exception trying to save ONNX graph. Please report this error on "
+                    "https://github.com/Unity-Technologies/ml-agents/issues and "
+                    "attach a copy of frozen_graph_def.pb"
+                )
+
+    else:
+        if _enforce_onnx_conversion():
+            raise RuntimeError(
+                "ONNX conversion enforced, but couldn't import dependencies."
```

Comment:
This is not true, it could also be because the version does not match.

Reply:
Good point, I'll update the comment here. It's not something a user will ever see (unless they added the environment variable), though.